sklearn/linear_model/least_angle.py

```diff
@@ -132,6 +132,243 @@ def lars_path(X, y, Xy=None, Gram=None, max_iter=500,
     LarsCV
     sklearn.decomposition.sparse_encode
 
+    References
+    ----------
+    .. [1] "Least Angle Regression", Effron et al.
+           http://statweb.stanford.edu/~tibs/ftp/lars.pdf
+
+    .. [2] `Wikipedia entry on the Least-angle regression
+           <https://en.wikipedia.org/wiki/Least-angle_regression>`_
+
+    .. [3] `Wikipedia entry on the Lasso
+           <https://en.wikipedia.org/wiki/Lasso_(statistics)>`_
+
+    """
+    return _lars_path_solver(
+        X=X, y=y, Xy=Xy, Gram=Gram, n_samples=None, max_iter=max_iter,
```

Comment:
you should deprecate the option to pass Xy and Gram to lasso_path. There should be one way to do things. Otherwise it makes maintenance and documentation harder for little gain.